### PR TITLE
Bugfix [CHK-3566] solved sorting on PSP list using string values

### PR DESCRIPTION
**Pull request by EmanueleBVtech**
#### List of Changes
Minor edit to fix sorting on PSP using names

#### Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#### Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: src/utils/SortUtil.ts [hunk 1]
```diff
@@ -17,5 +17,11 @@ export const sortBy =
       return -1; // Place undefined at the end
     }
 
-    return fieldA > fieldB ? order : -order;
+    // Check if both values are strings
+    if (typeof fieldA === "string" && typeof fieldB === "string") {
+      return fieldA.localeCompare(fieldB) * order;
```
Comment (pietro-tota) [inline src/utils/SortUtil.ts:22]: please extends test cases for string comparison with lower/upper case mixed cases in order to have a test to check comparison, maybe reporting the same names for psps that originated the bug

Comment (EmanueleBVtech) [reply src/utils/SortUtil.ts:22]: done here c1eac42055f0c707986e064e4667bdfe2f41a815

Comment (scaminati-bv) [inline src/utils/SortUtil.ts:22]: By default **localCompare** does not guarantee case insensitive ordering.
According to [this](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/Collator/Collator#sensitivity) use **base** option or if you prefear you can compare the two strings after toLowerCase()/toUpperCase() convertion

Comment (EmanueleBVtech) [reply src/utils/SortUtil.ts:22]: We should not use toLowerCase/toUpperCase for the following reason: localeCompare is the recommended approach when sorting JS arrays as it ensures that your sorting respects the nuances of different languages and character sets 
(takes into account the locale).

Added base fix tho.

Comment (scaminati-bv) [reply src/utils/SortUtil.ts]: I suggested using loweCase/upperCase with the localCompare function as-is.
`fieldA.toLowerCase().localeCompare(fieldB.toLowerCase())`.
BTW with parameter as you did is the best choice.

Comment (EmanueleBVtech) [reply src/utils/SortUtil.ts]: this is probably the absolute best
`fieldA.toLowerCase().localeCompare(fieldB.toLowerCase())`
but at this point i think we can do fine with the current code, right?

Comment (scaminati-bv) [reply src/utils/SortUtil.ts]: For me is equivalent, so use the version you prefear

Comment (EmanueleBVtech) [reply src/utils/SortUtil.ts]: if you are convinced, can you approve the PR?

**Review by scaminati-bv — APPROVED** [review]

**Review by ciuffagianluca — APPROVED** [review]

**Pull request merged**